tests/common/platform/processes_utils.py

```diff
@@ -0,0 +1,49 @@
+"""
+Helper script for checking status of critical processes
+
+This script contains re-usable functions for checking status of critical services.
+"""
+import logging
+import time
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+
+def _get_critical_processes_status(dut):
+    processes_status = dut.all_critical_process_status()
+    for k, v in processes_status.items():
+        if v['status'] == False or len(v['exited_critical_process']) > 0:
+            return False, processes_status
+
+    return True, processes_status
+
+def _all_critical_processes_healthy(dut):
+    logging.info("Check critical processes status")
+    status, _ = _get_critical_processes_status(dut)
+    return status
+
+def check_critical_processes(dut, watch_secs=0):
+    """
+    @summary: check all critical processes. They should be all running.
+              keep on checking every 5 seconds until watch_secs drops below 0.
+    @param dut: The AnsibleHost object of DUT. For interacting with DUT.
+    @param watch_secs: all processes should remain healthy for watch_secs seconds.
+    """
+    logging.info("Check all critical processes are healthy for {} seconds".format(watch_secs))
+    while watch_secs >= 0:
+        status, details = _get_critical_processes_status(dut)
+        pytest_assert(status, "Not all critical processes are healthy: {}".format(details))
```

Comment:
shouldn't this be just logging error since we want this loop to continue?

Reply:
The purpose of this loop is to make sure that there is no critical process failure for 60 seconds. If there is a failure, then the test should fail.

Reply:
What you have in mind is the other method: wait_critical_process()

Reply:
I got confused with the wait_critical_process approach. It is clear now